Mathlib/Data/List/Basic.lean

```diff
@@ -3898,6 +3898,10 @@ theorem nthLe_enum (l : List α) (i : ℕ) (hi' : i < l.enum.length)
     l.enum.nthLe i hi' = (i, l.nthLe i hi) := get_enum ..
 #align list.nth_le_enum List.nthLe_enum
 
+@[simp]
+theorem enum_eq_nil {l : List α} : List.enum l = [] ↔ l = [] := by
+  cases l <;> simp [List.enum_cons]
```

Comment:
The other `enum` theorems are proved in terms of an `enumFrom` lemma. Can you write that lemma for this case too?